File: app/src/main/java/com/jenni/jeezyfashion/di/login/LoginViewModelsModule.java
package com.jenni.jeezyfashion.di.login;

import androidx.lifecycle.ViewModel;

import com.jenni.jeezyfashion.di.ViewModelKey;
import com.jenni.jeezyfashion.ui.login.LoginViewModel;

import dagger.Binds;
import dagger.Module;
import dagger.multibindings.IntoMap;

@Module
public abstract class LoginViewModelsModule {
    @Binds
    @IntoMap
    @ViewModelKey(LoginViewModel.class)
    public abstract ViewModel bindLoginViewModel(LoginViewModel viewModel);
}

File: app/src/main/java/com/jenni/jeezyfashion/di/login/LoginViewModelsModule.kt
package com.jenni.jeezyfashion.di.login

import dagger.Binds
import dagger.multibindings.IntoMap
import com.jenni.jeezyfashion.di.ViewModelKey
import com.jenni.jeezyfashion.ui.login.LoginViewModel
import androidx.lifecycle.ViewModel
import dagger.Module

@Module
abstract class LoginViewModelsModule {
    @Binds
    @IntoMap
    @ViewModelKey(LoginViewModel::class)
    abstract fun bindLoginViewModel(viewModel: LoginViewModel?): ViewModel?
}